src/playground-project.ts

```diff
@@ -313,6 +336,8 @@ export class PlaygroundProject extends LitElement {
         source as void; // Exhaustive check.
         break;
     }
+    this._pristineFiles =
+      this._files && JSON.parse(JSON.stringify(this._files));
```

Comment:
Is there a more memory efficient way to ensure a file changed?  Also it probably will never get so big to matter.